elfpy/utils/apeworx_integrations.py

```diff
@@ -400,47 +400,44 @@ def get_wallet_from_trade_history(
         # check if there's an outstanding balance
         if balance != 0 or on_chain_balance != 0:
             if asset_type == "SHORT":
-                # loop across all the positions owned by this wallet
-                sum_product_of_open_share_price_and_value, sum_value = 0, 0
-                for specific_trade in relevant_trades.index[relevant_trades]:
-                    value = trade_history.loc[specific_trade, "value"]
-                    value *= -1 if trade_history.loc[specific_trade, "from"] == address else 1
-                    sum_value += value
-                    sum_product_of_open_share_price_and_value += (
-                        value * trade_history.loc[specific_trade, "share_price"]
-                    )
-                open_share_price = int(sum_product_of_open_share_price_and_value / sum_value)
-                assert abs(balance - sum_value) <= tolerance, "weighted average open share price calculation is wrong"
-                logging.debug("calculated weighted average open share price of %s", open_share_price)
                 previous_balance = wallet.shorts[mint_time].balance if mint_time in wallet.shorts else 0
-                new_balance = previous_balance + FixedPoint(scaled_value=balance)
+                delta_balance = FixedPoint(scaled_value=balance)
+                new_balance = previous_balance + delta_balance
                 if new_balance == 0:
-                    # remove key of "mint_time" from the "wallet.shorts" dict
                     wallet.shorts.pop(mint_time, None)
-                else:  # position balance is not zero
-                    wallet.shorts.update(
-                        {
-                            mint_time: Short(
-                                balance=new_balance,
-                                open_share_price=FixedPoint(scaled_value=open_share_price),
-                            )
-                        }
-                    )
-                    logging.debug(
-                        "storing in wallet as %s",
-                        {
-                            mint_time: Short(
-                                balance=new_balance, open_share_price=FixedPoint(scaled_value=open_share_price)
+                else:  # we either do a marginal update or full weighted average for open share price
+                    if add_to_existing_wallet:  # marginal update
```

Comment:
Can you summarize the logic in here? I'm confused as to why we are ok with setting the `previous_share_price` to 0.

Reply:
that denotes the case where the previous balance is also 0, so the new weighted average = `delta_price * delta_amount + old_balance * old_price` where the value of old_price doesn't matter. I'll see if I can draw this out more clearly.